README.md

```diff
@@ -0,0 +1,18 @@
+# Brigade Foundations
+
+Brigade Foundations is a Go module containing a number of foundational packages
+that are broadly used throughout various Brigade projects.
```

Comment:
I thought about it and initially opted not to just because this is geared for internal consumption. Willing to change course on that, but would that be ok in a follow-up?